### ISSUE TITLE
Why I cannot toggle on annotations?

### ISSUE THREAD

**Issue by fishfree** [Daniel-KM/Omeka-S-module-Mirador#3]
On [this page](https://deansforum.shisu.edu.cn/s/annotation/item/1), when I click on the "toggle annotations" icon,  no toolbar displays.
![noannotationsbar](https://user-images.githubusercontent.com/1741341/68529411-7938b480-0339-11ea-97f8-81812ff46cc3.jpg)

**Comment by Daniel-KM** [comment]
Do you want to display it or to remove it? By default, the annotations are useless, they require a backend, so it's better to remove the icon. See the readme for that.

**Comment by fishfree** [comment]
@Daniel-KM I want to add annotations.  I tried to add config JSON in the Mirador json config (item) 
section, admin/site/s/{mysite}/settings page as below:
`{
    "mainMenuSettings": {
        "show": false
    },
    "windowObjects": [
        {
            "loadedManifest": {manifestUri},
            "canvasID": {canvasID},
            "viewType": "ImageView",
            "displayLayout": false,
            "bottomPanel": false,
            "sidePanel": false,
            "annotationLayer": true,
            "canvasControls": {
                "annotations": true,
                "imageManipulation": {
                    "manipulationLayer": true
                }
            }
        }
    ]
}`

Still no luck with anonymous user or admin user.

So how to make annotation feature working? I know some annotation store server like [simpleannotationserver ](https://github.com/glenrobson/SimpleAnnotationServer)or [Harvard catch](https://github.com/annotationsatharvard/catcha).

**Comment by Daniel-KM** [comment]
Note that the config should be : 
```
             "loadedManifest": __manifestUri__,
             "canvasID": __canvasID__,
```
because `__manifestUri__` and `__canvasID__` are replaced statically.
But in a future version, it may be possible to use your config, it's cleaner.

For the annotation server, I didn't tried yet, so I don't know.

**Comment by fishfree** [comment]
@Daniel-KM I tried your spelling before posting this issue, no luck. Thank you all the same! :-)

**Comment by Daniel-KM** [comment]
Check https://forum.omeka.org/t/omeka-s-v2-0-2-with-iiif-and-mirador-support/9547 too.

**Comment by Daniel-KM** [comment]
And in the log, there is a message that is added when the json config is not good.

**Comment by fishfree** [comment]
@Daniel-KM Thanks! I followed [the tip ](https://forum.omeka.org/t/omeka-s-v2-0-2-with-iiif-and-mirador-support/9547/2)and now I can annotate. But annotation data can only saved in the browser's local storage. I want to it presistent? How to make that, please?
Edit: I notice you also develop the [Omeka-S-module-Annotate module](https://github.com/Daniel-KM/Omeka-S-module-Annotate), why cannot just use that module as the Mirador's annotation storage endpoint?

**Comment by Daniel-KM** [comment]
Yes, it's an important improvement to make Annotate an endpoint for annotation. Annotate is only a backend module. A specific module should be written to adapt the front end data to the back end. 

In fact, the Open Annotation standard has three parts: Data Model, Vocabulary and Protocol. Only the first is implemented, and it is enough for Omeka. The protocol should be implemented to support any other widget compatible with Open Annotation, but it is an heavy task, unless if a php library can be used (it wasn't the case when the module was created).

So for now, use another endpoint.

**Comment by fishfree** [comment]
@Daniel-KM Thank you for detailed explanation! Could you please tell me more about how to use another endpoint? Do you think [this repo & doc](https://github.com/annotationsatharvard/catcha/blob/7632c32b9026672d3257b58082410b46187e8e3c/docs/user-guide/annotatorjs.md) can be integrated?

**Comment by Daniel-KM** [comment]
It's not the client part that we need, because Mirador has already one, but the server part. So catcha or hypothesis can be used as server part. For this module, I need a php server endpoint part to integrate it fully.

**Comment by fishfree** [comment]
@Daniel-KM Daniel, I find the [SimpleAnnotationServer ](https://github.com/glenrobson/SimpleAnnotationServer/blob/master/doc/Endpoints.md)here, which provides HTTP API endpoints. So would you please integrate it with Omeka-S-module-Mirador to persist user-generated annotations?

**Comment by Daniel-KM** [comment]
For that, some code should be added in mirador as indicated in the [mirador-annotations](https://github.com/ProjectMirador/mirador-annotations) page. 

Currently, only the server annotot and the local storage are available. So this is a work to do, but it is a real work.

I don't understand why mirador need a specific connector for each endpoint, because it is the same protocol. So I think that a copy of the Annotot adapter with a different name "StandardAdapter" should be nearly fine. Or even did you try with the AnnototAdapter? Have you a annotation endpoint to check it?

**Comment by fishfree** [comment]
@Daniel-KM Daniel, thank you for your explanation! Sorry I'm not a programmer :-(.

**Comment by fishfree** [comment]
Any update on this issue?

**Comment by Daniel-KM** [comment]
Annotations works fine in last version of mirador (v4).
